master/internal/api_trials.go

```diff
@@ -573,6 +575,114 @@ func (a *apiServer) TrialPreemptionSignal(
 	}
 }
 
+func (a *apiServer) GetTrialSearcherTrainUntil(
+	_ context.Context, req *apiv1.GetTrialSearcherTrainUntilRequest,
+) (*apiv1.GetTrialSearcherTrainUntilResponse, error) {
+	exp, err := a.experimentActorFromTrialID(int(req.TrialId))
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := a.askAtDefaultSystem(exp, trialTrainUntilReq{
+		trialID: int(req.TrialId),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	switch tResp, ok := resp.(trialTrainUntilResp); {
+	case !ok:
+		return nil, unexpectedMessageError(exp, resp)
+	case tResp.finished:
+		return nil, nil
+	default:
+		return &apiv1.GetTrialSearcherTrainUntilResponse{
+			Length: tResp.length.ToProto(),
+		}, nil
+	}
+}
+
+func (a *apiServer) ReportTrialSearcherValidation(
+	_ context.Context, req *apiv1.ReportTrialSearcherValidationRequest,
+) (*apiv1.ReportTrialSearcherValidationResponse, error) {
+	exp, err := a.experimentActorFromTrialID(int(req.TrialId))
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO(DET-5210): Sending a trial snapshot along forces an experiment snapshot,
+	// but with a nil snapshot it won't save the trial snapshot. At the end of push
+	// arch, we should just remove trial snapshots entirely (they should snapshotted
+	// but separately, not through/with experiments, since it's really just run id and restarts).
+	if _, err = a.askAtDefaultSystem(exp, trialReportValidation{
```

Comment:
yeah, mostly block until the validation is successfully reported instead of returning then the error only surfacing in master logs.

Reply:
Wait.  Why does the trial even need this?  It seems like it should go straight to the database, and the trial actor should be a consumer of this api rather than the engine behind it?

Reply:
> Why does the trial even need this? 

not sure what you mean. this reports the validation to the searcher through the experiment actor, the same way the trial actor does it now.

Reply:
Ooooh, I was reading `ReportTrialValidation`, did not realize this for searcher validations.  Ignore me...